edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/ModuleManagementHttpClient.cs

```diff
@@ -112,5 +115,33 @@ static ApiVersion GetSupportedVersion(string serverSupportedApiVersion, string c
 
             return serverVersion.Value < clientVersion.Value ? serverVersion : clientVersion;
         }
+
+        Task Throttle(Func<Task> identityOperation)
+        {
+            return this.Throttle<bool>(
+                            async () =>
+                            {
+                                await identityOperation();
+                                return true;
+                            });
+        }
+
+        async Task<T> Throttle<T>(Func<Task<T>> identityOperation)
+        {
+            bool permitAcquired = await this.clientPermit.WaitAsync(this.clientPermitTimeout);
```

Comment:
```suggestion
            using(bool permitAcquired = await this.clientPermit.WaitAsync(this.clientPermitTimeout)) {
```

This way you don't need the try-finally